### ISSUE TITLE
StrawberryShake Blazor component stuck on loading upon second page navigation.

### ISSUE THREAD

**Issue by rainbow-dolly** [ChilliCream/graphql-platform#5988]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Strawberry Shake

### Describe the bug

Hi! I started learning StrawberryShake and GraphQL abou a week ago and ran into a problem.
I am using Blazor WASM and after generating my client I set up a few test-pages that grab some data from my GraphQL server and display them in a table.
Basically using:
```
@page "/overviews/priorities"

<h1>Priorities</h1>

<MyFirstApp.Components.UseReadPriorities Context="priorities">
    <ChildContent>
        <table class="table">
        ...
        </table>
    </ChildContent>
    <ErrorContent>
        Something went wrong ...<br />
        @priorities.First().Message
    </ErrorContent>
    <LoadingContent>
        Loading ...
    </LoadingContent>
</MyFirstApp.Components.UseReadPriorities>
```
The problem that I have is, this works perfectly fine when I navigate to this page the first time, but after leaving the page and navigating to the page again it gets stuck on Loading. I see in the browser that the GraphQL request completes successfully but the component does not switch to Child Content.
This fixes itself after hitting Refresh in the browser, but that's obviously not desirable in an SPA.
Am I doing something wrong or is this a bug?

### Steps to reproduce

1. Create a new Blazor WASM Project in VS Code.
2. Add some graphql client with dotnet-graphql init + a query.
3. Add a new page to the default page navigation and use the generated StrawberryShake component on that page.
4. Launch the application.
5. Navigate to the page that holds the GraphQL component. The data should be displayed correctly.
6. Leave the page that holds the GraphQL component via navigation.
7. Navigate to the page that holds the GraphQL component again. The component is now stuck on Loading ...

### Version

13.0.5

**Comment by valincius** [comment]
+1 I am facing a similar issue

**Comment by A360JMaxxgamer** [comment]
I had a similiar issue too. 

I was able to track the cause of my problem. I guess it might be the same here.

The problem seems to be a threading issue. The generated components derive either from UseQuery or UseSubscription. Both of these components subscribe to updates of the observable from the operationstore. I faced the same problem a few years ago in WPF. The subscription callback is not necessary handled on the UI thread. So calling StateHasChanged does not lead to retrigger BuildRenderTree. 

```csharp
protected void Subscribe(IObservable<IOperationResult<TResult>> observable)
  {
      _subscription?.Dispose();

      _subscription = observable
          .Subscribe(operationResult =>
          {
              _result = operationResult.Data;
              _errors = operationResult.Errors;
              _isErrorResult = operationResult.IsErrorResult();
              _isSuccessResult = operationResult.IsSuccessResult();
              _isInitializing = false;
              StateHasChanged();
          });
  }
```

To ensure that StateHasChanged is triggered on the UI thread we need to call InvokeAsync(StateHasChanged) .

```csharp
protected void Subscribe(IObservable<IOperationResult<TResult>> observable)
    {
        _subscription?.Dispose();

        _subscription = observable
            .Subscribe(operationResult =>
            {
                _result = operationResult.Data;
                _errors = operationResult.Errors;
                _isErrorResult = operationResult.IsErrorResult();
                _isSuccessResult = operationResult.IsSuccessResult();
                _isInitializing = false;
                InvokeAsync(StateHasChanged);
            });
    }
```

That fixed the problem for me locally. 

I will prepare a pr right away and link this issue.

**Comment by JaredWalden00** [comment]
I also have encountered this same issue. I assume it as a race condition, as removing some of the data that would get loaded in the query caused it to load on second load up. I also found another round about which is to call OnParametersSetAsync, as the info is being dropped into the component, just not directly at the start of the second rendering. Honestly it is very odd and I'm not a huge fan of the round abouts I've had to use
